add a section for troubleshooting composition (#430)
crossplane/docs: content/knowledge-base/guides/troubleshoot.md

> Note that a reference to a `ControllerConfig` can be added to an already
> installed `Provider` and it will update its `Deployment` accordingly.

## Compositions and XRDs

### General troubleshooting steps

Most error messages are logged to resources' event fields. Whenever your Composite Resources are not getting provisioned, follow the following steps:

1. Get the events for the root resource using `kubectl describe` or `kubectl get event`
2. If there are errors in the events, address them.

> 3. If no errors, follow its sub-resources. `kubectl get <KIND> <NAME> -o=jsonpath='{.spec.resourceRef}{" "}{.spec.resourceRefs}' | jq`

Reviewer: ```suggestion
3. If there are no errors, follow its sub-resources.   
`kubectl get <KIND> <NAME> -o=jsonpath='{.spec.resourceRef}{" "}{.spec.resourceRefs}' | jq`
```

Puts the command on a new line

> 4. Go back to step 1 using one of resources returned by step 3.

Reviewer: ```suggestion
4. Repeat this process for each resource returned.
```

> The rest of this section will show you how to debug issues related to compositions without using external tooling. However, if you are using ArgoCD or FluxCD with UI, you can visualize object relationships more easily in the UI. You can also use the kube-lineage plugin to visualize object relationships in your terminal.

Reviewer: ```suggestion
{{< hint "note" >}}
The rest of this section will show you how to debug issues related to compositions without using external tooling. 

If you are using ArgoCD or FluxCD with UI, you can visualize object relationships in the UI. 

You can also use the kube-lineage plugin to visualize object relationships in your terminal.
{{< /hint >}}
```

Wrapping in a hint since it's information not directly relevant to the main content.

### Example

#### Composition
An example application was deployed as a claim of a composite resource. Kind = `ExampleApp`. Name = `example-application`.

The example application never reaches available state as shown below.

> 1. Run `kubectl describe exampleapp example-application`
> 
>     ```bash
>     Status:

Reviewer: ```suggestion
1. View the claim

    ```bash
kubectl describe exampleapp example-application
    Status:
```
Make the step apply to any setup and use the example output for your specific example.

    Conditions:
        Last Transition Time:  2022-03-01T22:57:38Z
        Reason:                Composite resource claim is waiting for composite resource to become Ready
        Status:                False
        Type:                  Ready
    Events:                    <none>
    ```

> 2. No errors were found in events of the claim. Find its cluster scoped resource (composite resource).

Reviewer: ```suggestion
2. If the claim doesn't have errors, inspect the composite resource
```
Be more direct and generic in the step

    ```bash
    kubectl get exampleapp example-application -o=jsonpath='{.spec.resourceRef}{" "}{.spec.resourceRefs}' | jq

    {
      "apiVersion": "awsblueprints.io/v1alpha1",
      "kind": "XExampleApp",
      "name": "example-application-xqlsz"
    }
    ```

> 3. In the above output, we see the cluster scoped resource for this claim. Kind = `XExampleApp` name = `example-application-xqlsz`
> 4. Get the cluster resource's event.

Reviewer: ```suggestion
4. If there are no errors, get the resource's event.
```
Simplify and make more step-by-step instructions.
